clang/lib/Driver/Driver.cpp

```diff
@@ -3766,6 +3766,20 @@ class OffloadingActionBuilder final {
         return SYCLDeviceOnly ? ABRT_Ignore_Host : ABRT_Success;
       }
 
+      if (CurPhase == phases::Preprocess) {
+        // Do not perform the host compilation when doing preprocessing only
+        // with -fsycl-device-only.
+        if (Args.hasArg(options::OPT_fsycl_device_only) &&
+            (Args.getLastArg(options::OPT__SLASH_EP, options::OPT__SLASH_P) ||
+             Args.getLastArg(options::OPT_E) ||
+             Args.getLastArg(options::OPT_M, options::OPT_MM))) {
```

Comment:
```suggestion
        bool IsPreprocessingOnly = Args.getLastArg(options::OPT_E) ||
             Args.getLastArg(options::OPT_M, options::OPT_MM) ||
             Args.getLastArg(options::OPT__SLASH_EP, options::OPT__SLASH_P);
        if (Args.hasArg(options::OPT_fsycl_device_only) && IsPreprocessingOnly) {
```
This way, the intent might be a tad clearer to the reader.